packages/material-ui/src/styles/experimentalStyled.d.ts

```diff
@@ -166,6 +166,7 @@ export interface CreateMUIStyled<Theme extends object = DefaultTheme> {
       theme?: Theme;
       as?: React.ElementType;
       sx?: SxProps<Theme>;
+      styleProps?: Record<string, any>;
```

Comment:
`styleProps` should be a part of StyledComponent so I added it. For the safe side, use `Record<string, any>` because styleProps varies across components

Reply:
Does `Record<string, unknown>` work as well?

Reply:
updated, it works.